xsnap/sources/xsnap-worker.c

```diff
@@ -878,17 +896,17 @@ static int fxWriteOkay(FILE* outStream, xsUnsignedValue meterIndex, xsMachine *t
 		tsbuf = "[]";
 	}
 	char fmt[] = ("." // OK
-				  "{"
-				  "\"currentHeapCount\":%u,"
-				  "\"compute\":%u,"
-				  "\"allocate\":%u,"
-				  "\"timestamps\":%s}"
-				  "\1" // separate meter info from result
-				  );
+					"{"
+					"\"currentHeapCount\":%u,"
+					"\"compute\":%u,"
+					"\"allocate\":%u,"
+					"\"timestamps\":%s}"
+					"\1" // separate meter info from result
+					);
 	char numeral64[] = "12345678901234567890"; // big enough for 64bit numeral
 	char prefix[8 + sizeof fmt + 8 * sizeof numeral64 + sizeof timestampBuffer];
 	// Prepend the meter usage to the reply.
-	snprintf(prefix, sizeof(prefix), fmt,
+	snprintf(prefix, sizeof(prefix) - 1, fmt,
```

Comment:
Do we need to make prefix 1 bigger instead?

Reply:
That’s reasonable. I believe `sprintfn` assumes a space for the null terminator beyond the length.

Reply:
>     The snprintf() and vsnprintf() functions will write at most size-1 of the characters
>     printed into the output string (the size'th character then gets the terminating ‘\0’); if
>     the return value is greater than or equal to the size argument, the string was too short
>     and some of the printed characters were discarded.  The output is always null-terminated,
>     unless size is 0.